samples/snippets/fsharp/azure/blob-storage.fsx

```diff
@@ -29,88 +25,67 @@ let storageAccount = CloudStorageAccount.Parse(storageConnString)
-//
-
-let blobClient = storageAccount.CreateCloudBlobClient()
-
 //
 // Create a container.
 //
 
- // Retrieve a reference to a container.
-let container = blobClient.GetContainerReference("mydata")
+ // Create a blob container client.
+let container = BlobContainerClient(storageConnString, "myContainer");
 
 // Create the container if it doesn't already exist.
 container.CreateIfNotExists()
 
-let permissions = BlobContainerPermissions(PublicAccess=BlobContainerPublicAccessType.Blob)
-container.SetPermissions(permissions)
+let permissions = PublicAccessType.Blob
+container.SetAccessPolicy(permissions)
 
 //
 // Upload a blob into a container.
 //
 
 // Retrieve reference to a blob named "myblob.txt".
-let blockBlob = container.GetBlockBlobReference("myblob.txt")
+let blockBlob = container.GetBlockBlobClient("myblob.txt")
 
 // Create or overwrite the "myblob.txt" blob with contents from the local file.
-do blockBlob.UploadFromFile(localFile)
-
-
+use fileStream = new FileStream(localFile, FileMode.Open, FileAccess.Read, FileShare.Read)
+let offset = 0
+let length = int(fileStream.Length)
+let bytes = Array.zeroCreate<byte>(length)
+fileStream.Read(bytes, offset, bytes.Length)
+fileStream.Close();
+use stream = new MemoryStream(bytes)
+do blockBlob.Upload(stream)
 
 //
 // List the blobs in a container.
 //
 
-// Loop over items within the container and output the length and URI.
-for item in container.ListBlobs(null, false) do
-    match item with 
-    | :? CloudBlockBlob as blob -> 
-        printfn "Block blob of length %d: %O" blob.Properties.Length blob.Uri
-
-    | :? CloudPageBlob as pageBlob ->
-        printfn "Page blob of length %d: %O" pageBlob.Properties.Length pageBlob.Uri
-
-    | :? CloudBlobDirectory as directory ->
-        printfn "Directory: %O" directory.Uri
-
-    | _ ->
-        printfn "Unknown blob type: %O" (item.GetType())
-
-// Loop over items within the container and output the length and URI.
-for item in container.ListBlobs(null, true) do
-    match item with 
-    | :? CloudBlockBlob as blob -> 
-        printfn "Block blob of length %d: %O" blob.Properties.Length blob.Uri
-
-    | _ ->
-        printfn "Unexpected blob type: %O" (item.GetType())
+// Loop over items within the container and output the name.
+for item in container.GetBlobs() do
+    printfn $"Blob name: {item.Name}"
 
 //
 // Download Blobs.
 //
 
 // Retrieve reference to a blob named "myblob.txt".
-let blobToDownload = container.GetBlockBlobReference("myblob.txt")
+let blobToDownload = container.GetBlobClient("myblob.txt")
 
 // Save blob contents to a file.
 do
     use fileStream = File.OpenWrite(__SOURCE_DIRECTORY__ + "/path/download.txt")
-    blobToDownload.DownloadToStream(fileStream)
+    blobToDownload.DownloadTo(fileStream)
 
 let text =
     use memoryStream = new MemoryStream()
-    blobToDownload.DownloadToStream(memoryStream)
+    blobToDownload.DownloadTo(memoryStream)
     Text.Encoding.UTF8.GetString(memoryStream.ToArray())
```

Comment:
Prefer `let text = blobToDownload.DownloadContent().Content.ToString()` which does the same thing